src/main/java/org/folio/innreach/batch/contribution/service/ContributionJobRunner.java

```diff
@@ -64,57 +73,125 @@ public class ContributionJobRunner {
-          var iterationJobId = context.getIterationJobId();
-
-          if (isUnknownEvent(event, iterationJobId)) {
-            log.info("Skipping unknown event, current job is {}", iterationJobId);
-            continue;
-          }
-
-          var instance = loadInstanceWithItems(instanceId);
-          if (instance == null) {
-            continue;
-          }
-
-          if (isEligibleForContribution(centralServerId, instance)) {
-            contributeInstance(centralServerId, instance, stats);
-            contributeInstanceItems(centralServerId, instance, stats);
-          } else if (isContributed(centralServerId, instance)) {
-            deContributeInstance(centralServerId, instance, stats);
-          }
-        }
-      });
+
+    var iterationJobId = context.getIterationJobId();
+
+    if (event == null) {
+      log.info("Exiting kafka message reader for contribution {} and job {}", contributionId, iterationJobId);
+      return;
+    }
+    log.info("Processing instance iteration event = {}", event);
+
+    var instanceId = event.getInstanceId();
+
+    if (isUnknownEvent(event, iterationJobId)) {
+      log.info("Skipping unknown event, current job is {}", iterationJobId);
+      return;
+    }
+
+    Instance instance = loadInstanceWithItems(instanceId);
+
+    if (instance == null) {
+      log.info("Instance is null, skipping");
+      return;
+    }
+
+    var centralServerId = context.getCentralServerId();
+
+    if (isEligibleForContribution(centralServerId, instance)) {
+      contributeInstance(centralServerId, instance, stats);
+      contributeInstanceItems(centralServerId, instance, stats);
+    } else if (isContributed(centralServerId, instance)) {
+      deContributeInstance(centralServerId, instance, stats);
     }
+    if (Objects.equals(recordsProcessed.get(context.getTenantId()), totalRecords)) {
+      completeContribution(context, stats);
+      endContributionJobContext();
+      InitialContributionJobConsumerContainer.stopConsumer(topic);
+    }
+  }
+
+  // TODO Try to simulate the stats collection. Remove after testing.
+  public void simulateContribution() {
+    makeSimulatedRequest();
+//    s.addRecordsTotal(1);
+  }
+
+  private void simulateContributionItems() {
+    // For now just make this one.
+    makeSimulatedRequest();
+//    s.addRecordsTotal(1);
+  }
+
+  private void makeSimulatedRequest(Statistics s) {
+    // Simulate the try/catch around http client calls.
+    try {
+      log.info("Making simulated request");
+      String postBody = "somerandomstring"; // Should probably make this a random length.
+      URI testServer = URI.create("http://localhost:8080");
+      var res = testClient.makeTestRequest(testServer, postBody);
+      log.info("Response from test server: {}", res);
+    } catch (Exception e) {
+      log.warn("Error while simulating request: {} {}", e.getMessage(), e.getStackTrace());
+    } finally {
+      s.addRecordsProcessed(1);
+    }
+  }
```

Comment:
I think this method can be removed. It's not useful anymore is it? We have the one below which we can keep.